pkg/storegateway/bucket.go

```diff
@@ -451,6 +451,10 @@ func (s *BucketStore) addBlock(ctx context.Context, meta *block.Meta) (err error
 			level.Error(s.logger).Log("msg", "loading block failed", "elapsed", time.Since(start), "id", meta.ULID, "err", err)
 		} else {
 			level.Info(s.logger).Log("msg", "loaded new block", "elapsed", time.Since(start), "id", meta.ULID)
+
+			// Record block discovery latency as time from block creation (ULID timestamp) to now.
+			blockCreationTime := time.UnixMilli(int64(meta.ULID.Time()))
+			s.metrics.blockDiscoveryLatency.Observe(time.Since(blockCreationTime).Seconds())
```

Comment:
won't this create a lot of noise when a store-gateway restarts? it will record this metric for all blocks it owns. It will make it hard for us to detect slowdowns in the block shipping&discovery process

Reply:
my idea was to somehow do this only for blocks which have not been loaded before. perhaps we can do it while downloading the index header or something